src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,242 @@
+                                + contentElement.getClass().getName()
+                                + "] cannot be cast to ["
+                                + String.class.getName()
+                                + "]"
+                        );
+                    }
+                }
+            } else if (!(content instanceof String)) {
+                throw new IllegalArgumentException(
+                    "input field ["
+                        + inputField
+                        + "] of type ["
+                        + content.getClass().getName()
+                        + "] cannot be cast to ["
+                        + String.class.getName()
+                        + "]"
+                );
+            }
+
+            @SuppressWarnings("unchecked")
+            Map<String, Object> parameters = (Map<String, Object>) fieldMapEntry.getValue();
+            String outputField = (String) parameters.get(OUTPUT_FIELD);
+            List<String> chunkedPassages = new ArrayList<>();
+
+            // we have validated that there is one chunking algorithm
+            // and that chunkerParameters is of type Map<String, Object>
+            for (Map.Entry<String, Object> parameterEntry : parameters.entrySet()) {
+                String parameterKey = parameterEntry.getKey();
+                if (supportedChunkers.contains(parameterKey)) {
+                    @SuppressWarnings("unchecked")
+                    Map<String, Object> chunkerParameters = (Map<String, Object>) parameterEntry.getValue();
+                    if (Objects.equals(parameterKey, ChunkerFactory.FIXED_LENGTH_ALGORITHM)) {
+                        // add maxTokenCount to chunker parameters
+                        Map<String, Object> sourceAndMetadataMap = document.getSourceAndMetadata();
+                        int maxTokenCount = IndexSettings.MAX_TOKEN_COUNT_SETTING.get(settings);
+                        String indexName = sourceAndMetadataMap.get(IndexFieldMapper.NAME).toString();
+                        IndexMetadata indexMetadata = clusterService.state().metadata().index(indexName);
+                        if (indexMetadata != null) {
+                            // if the index exists, read maxTokenCount from the index setting
+                            IndexService indexService = indicesService.indexServiceSafe(indexMetadata.getIndex());
+                            maxTokenCount = indexService.getIndexSettings().getMaxTokenCount();
+                        }
+                        chunkerParameters.put(FixedTokenLengthChunker.MAX_TOKEN_COUNT, maxTokenCount);
+                    }
+                    IFieldChunker chunker = ChunkerFactory.create(parameterKey, analysisRegistry);
+                    if (content instanceof String) {
+                        chunkedPassages = chunker.chunk((String) content, chunkerParameters);
+                    } else {
+                        for (Object contentElement : (List<?>) content) {
```

Comment:
I think here for a list of string as input, the output should be List<List<String> >, otherwise, customer can not figure out which part of list belongs to which input string

Reply:
IMO, instead of `[["a", "b"], ["c", "d"]]`, we should return a flattened list like `["a", "b", "c", "d"]`. Please correct me if I am wrong @model-collapse.